### PR TITLE
Start DHCP Relay When Helpers IPs Are Available

### PR DESCRIPTION
**Pull request by tahmed-dev**
#### Why I did it

It is possible to have DHCP relay configuration with no servers/
helpers which result in DHCP container to crash. This PR fixes this
issue by not starting DHCP relay for vlans with no DHCP helpers.

signed-of-by: Tamer Ahmed <tamer.ahmed@microsoft.com>

resolves: #6931 
closes: #6931 
#### How I did it
Do not add program group for dhcp relay with not dhcp helpers

#### How to verify it
Unit test

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [x] 201911
- [ ] 202006
- [x] 202012

### CODE REVIEW

File: dockers/docker-dhcp-relay/docker-dhcp-relay.supervisord.conf.j2 [hunk 1]
```diff
@@ -43,7 +43,7 @@ dependent_startup_wait_for=rsyslogd:running
 {# Count how many VLANs require a DHCP relay agent... #}
 {% set num_relays = { 'count': 0 } %}
 {% for vlan_name in VLAN_INTERFACE %}
-{% if VLAN and vlan_name in VLAN and VLAN[vlan_name]['dhcp_servers'] %}
+{% if VLAN and vlan_name in VLAN and VLAN[vlan_name]['dhcp_servers']|length > 0 %}
```
Comment (qiluo-msft) [inline dockers/docker-dhcp-relay/docker-dhcp-relay.supervisord.conf.j2:46]: >dhcp_servers [](start = 54, length = 12)

Even better, check `'dhcp_servers' in VLAN[vlan_name]`

**Review by qiluo-msft — COMMENTED** [review]
As comment

**Review by lguohan — APPROVED** [review]

File: dockers/docker-dhcp-relay/docker-dhcp-relay.supervisord.conf.j2 (hunk 1 above)
Comment (tahmed-dev) [reply dockers/docker-dhcp-relay/docker-dhcp-relay.supervisord.conf.j2]: done

**Review by qiluo-msft — APPROVED** [review]
Thanks!

**Comment by abdosi** [comment]
@tahmed-dev please create PR for 201911.
@qiluo-msft

**Pull request merged**